ignite/contrib/handlers/param_scheduler.py

```diff
@@ -8,28 +8,38 @@ class ParamScheduler(object):
     training.
 
     Args:
-        optimizer (`torch.optim.Optimizer`): the optimizer to use
+        optimizer (`torch.optim.Optimizer`/dict): the optimizer or parameters group to use
         param_name (str): name of optimizer's parameter to update
         save_history (bool, optional): whether to log the parameter values
             (default=False)
     """
     def __init__(self, optimizer, param_name, save_history=False):
+
         self.optimizer = optimizer
+        if isinstance(optimizer, dict):
+            self.optimizer_param_groups = [optimizer]
+        else:
+            self.optimizer_param_groups = optimizer.param_groups
         self.param_name = param_name
         self.save_history = save_history
         self.event_index = 0
 
-    def __call__(self, engine):
+    def __call__(self, engine, name=None):
+
         value = self.get_param()
-        for param_group in self.optimizer.param_groups:
+
+        for param_group in self.optimizer_param_groups:
             param_group[self.param_name] = value
 
+        if name is None:
+            name = self.param_name
+
         if self.save_history:
             if not hasattr(engine.state, 'param_history'):
                 setattr(engine.state, 'param_history', {})
-            engine.state.param_history.setdefault(self.param_name, [])
-            values = [pg[self.param_name] for pg in self.optimizer.param_groups]
-            engine.state.param_history[self.param_name].append(values)
+            engine.state.param_history.setdefault(name, [])
```

Comment:
Just wonder whether it should not be 
```
if not hasattr(engine.state, 'param_history'):
    setattr(engine.state, 'param_history', {})
    engine.state.param_history.setdefault(name, [])
```

Reply:
I think not. `param_history` might already exist (added by another scheduler), but `name` can still be missing.